compiler/rustc_arena/src/lib.rs

```diff
@@ -114,6 +117,102 @@ impl<T> Default for TypedArena<T> {
     }
 }
 
+#[inline]
+unsafe fn write_from_iter<'a, T, I: Iterator<Item = T>>(
+    mut iter: I,
+    len: usize,
```

Comment:
Can we call this "expected_len" or something like that? Otherwise it seems a bit confusing to see a len that's only a hint, really.